src/components/app/App.js

```diff
@@ -1,28 +1,43 @@
 import React, { Component } from 'react';
 import Posters from '../posters/Posters';
 import MovieDetails from '../movie_details/MovieDetails';
-import movieData from '../../movieData';
+import ErrorCode from '../error_code/ErrorCode';
+import { getApiData } from '../../apiCalls';
 import './App.css';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movies: movieData.movies,
-      selectedMovie: null
+      movies: [],
+      errorCode: null,
+      selectedMovie: null,
+      fetchingMovie: null
     }
   }
 
-  selectMovie = (event) => {
-    const match = this.state.movies.find(movie => movie.id === parseInt(event.target.id));
-    
-    window.scrollTo(0,0);
+  componentDidMount = async () => {
+    try {
+      const fetchedMovies = await getApiData('movies');
+      this.setState({movies: fetchedMovies.movies});
+    } catch (e) {
+      this.setState({errorCode: e.message})
+    }
+  }
 
-    this.setState({ selectedMovie: match });
+  selectMovie = async (event) => {
+    try {
+      this.setState({fetchingMovie: true});
+      const match = await getApiData(`movies/${event.target.id}`);
+      window.scrollTo(0,0);
+      this.setState({selectedMovie: match.movie});
+    } catch (e) {
+      this.setState({errorCode: e.message})
+    }
   }
 
   clearSelected = () => {
-    this.setState({ selectedMovie: null });
+    this.setState({errorCode: null, selectedMovie: null, fetchingMovie: null});
```

Comment:
nice! Love to see one call of `setState` to adjust multiple pieces

Reply:
cool. Good to know! I was worried this was doing too much and we'd get the opposite feedback.